src/docs/development/platform-integration/add-to-app-android/add-splash-screen.md

```diff
@@ -0,0 +1,221 @@
+</style>
+```
+
+### Setup FlutterActivity in AndroidManifest.xml
+
+In `AndroidManifest.xml`, set the `theme` of `FlutterActivity` to the launch
+theme. Then, add a meta-data element to the desired `FlutterActivity` to instruct
+Flutter to switch from the launch theme to normal theme at the appropriate time.
+
+```xml
+<activity
+    android:name=".MyActivity"
+    android:theme="@style/LaunchTheme"
+    // ...
+    >
+    <meta-data
+        android:name="io.flutter.embedding.android.NormalTheme"
+        android:resource="@style/NormalTheme"
+        />
+    <intent-filter>
+        <action android:name="android.intent.action.MAIN"/>
+        <category android:name="android.intent.category.LAUNCHER"/>
+    </intent-filter>
+</activity>
+```
+
+The Android app now displays the desired launch screen while the app
+initializes.
+
+## Flutter Splash Screen
+
+Each Flutter experience in an app requires a few moments to initialize the Dart
+isolate that runs the code. This means a user momentarily sees a blank screen.
+Flutter supports an improved user experience by displaying an Android `View` as 
+a splash screen while Flutter initializes.
+
+Flutter supports two options for a splash screen. Developers can choose to
+display a `Drawable` of their choice, which then fades out once initialization
+is complete. The other option is to provide a custom `SplashScreen`, which is
+capable of displaying any Android `View` content that is desired.
+
+### Showing a Drawable splash screen
+
+A `Drawable` splash screen can be configured for a `FlutterActivity`,
+`FlutterFragment`, or `FlutterView`.
+
+#### In a FlutterActivity
+
+To display a `Drawable` as a Flutter splash screen in a `FlutterActivity`, add
+the following meta-data to the associated `FlutterActivity` in
+`AndroidManifest.xml`.
+
+```xml
+<meta-data
+    android:name="io.flutter.embedding.android.SplashScreenDrawable"
+    android:resource="@drawable/my_splash"
+    />
+```
+
+#### In a FlutterFragment
+
+To display a `Drawable` as a Flutter splash screen in a `FlutterFragment`,
+subclass `FlutterFragment` and override `provideSplashScreen()`.
+
+```java
+public class MyFlutterFragment extends FlutterFragment {
+    @Override
+    protected SplashScreen provideSplashScreen() {
+        // Load the splash Drawable.
+        Drawable splash = getResources().getDrawable(R.drawable.my_splash);
+
+        // Construct a DrawableSplashScreen with the loaded splash Drawable and
+        // return it.
+        return new DrawableSplashScreen(splash);
+    }
+}
+```
+
+#### With a FlutterView
+
+TODO(mattcarroll)
+
+### Creating a custom SplashScreen
+
+Splash screens are a great branding opportunity. As such, many teams choose to
+implement unique, highly-customized splash experiences. To facilitate this,
+Flutter allows developers to display an arbitrary Android `View` as a splash
+screen, and even allows developers to control how that `View` transitions to
+Flutter.
```

Comment:
transitions to Flutter once Flutter has its first frame ready and rendered.

Reply:
done